src/deep-security/lib/CredentialsManager.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Created by mgoria on 11/6/15.
+ */
+
+'use strict';
+
+import AWS from 'aws-sdk';
+import CognitoSyncManager from 'amazon-cognito-js';
+import {CreateCognitoDatasetException} from './Exception/CreateCognitoDatasetException';
+import {PutCognitoRecordException} from './Exception/PutCognitoRecordException';
+import {SynchronizeCognitoDatasetException} from './Exception/SynchronizeCognitoDatasetException';
+
+export class CredentialsManager {
+  /**
+   * @returns {String}
+   */
+  static get DATASET_NAME() {
+    return 'deep_session';
+  }
+
+  /**
+   * @returns {String}
+   */
+  static get RECORD_NAME() {
+    return 'session_creds';
+  }
+
+  /**
+   * @param {String} identityPoolId
+   * @param {Object|null} cognitoSyncClient
+   */
+  constructor(identityPoolId, cognitoSyncClient = null) {
+    this._identityPoolId = identityPoolId;
+    this._cognitoSyncClient = cognitoSyncClient;
+  }
+
+  /**
+   * @returns {Object}
+   */
+  get cognitoSyncClient() {
+    if (!this._cognitoSyncClient) {
+      this._cognitoSyncClient = new AWS.CognitoSyncManager();
+    }
+
+    return this._cognitoSyncClient;
+  }
+
+  /**
+   * @param {Object} credentials
+   * @param {Function} callback
+   */
+  saveCredentials(credentials, callback) {
+    this._createOrGetDataset((error, dataset) => {
+      if (error) {
+        return callback(error);
```

Comment:
Please do not return a callback result since it could create unexpected behaviour. use `return;` explicitly after calling the cb.